tests/Tests/Mapping/LegacyReflectionFieldsTest.php

```diff
@@ -56,20 +60,32 @@ public function testGetSet(): void
 
     public function testGetSetReadonly(): void
     {
-        $class = $this->dm->getClassMetadata(Tag::class);
+        $class = $this->dm->getClassMetadata(ReadOnlyProperty::class);
         self::assertInstanceOf(LegacyReflectionFields::class, $class->reflFields);
 
-        $tag = new Tag('Important');
+        $tag = new ReadOnlyProperty('Important');
         $this->dm->persist($tag);
         $this->dm->flush();
 
-        $tag = $this->dm->find(Tag::class, $tag->id);
+        $tag = $this->dm->find(ReadOnlyProperty::class, $tag->id);
 
         // Accessing the readonly property through reflection
         self::assertEquals('Important', $class->getReflectionProperty('name')->getValue($tag));
 
         self::expectException(LogicException::class);
-        self::expectExceptionMessage('Attempting to change readonly property Documents\Tag::$name');
+        self::expectExceptionMessage(sprintf('Attempting to change readonly property %s::$name', ReadOnlyProperty::class));
         $class->getReflectionProperty('name')->setValue($tag, 'Very Important');
     }
 }
+
+#[Document]
+class ReadOnlyProperty
+{
+    #[Id]
+    public string $id;
+
+    public function __construct(#[Field]
+    public readonly string $name,)
```

Comment:
Something weird happened here 🤨 

```suggestion
    public function __construct(
        #[Field]
        public readonly string $name
    )
```

Reply:
The way this happened is a mystery.